### PR TITLE
Update UG: Improve from PED

### LINKED ISSUE #192: [PE-D][Tester A] A person can be both guest and vendor at the same time.
In my opinion, I think a person can only either be a guest or a vendor, but not both at the same time. This could be handled, or at least state whether a person could be both guest and vendor at the same time or not to minimise ambiguity. 

1. Expected: A person to be either a guest or a vendor, but not both at the same time.
2. Actual: A person can be both guest and vendor at the same time.

Below is a screenshot of this minor bug:

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: KamJiaYue/ped#1

### LINKED ISSUE #209: [PE-D][Tester E] Clear command missing from UG
Clear works on your system, but not added in the UG. 

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: neyapraveen/ped#4

### LINKED ISSUE #214: [PE-D][Tester E] Address accepts single integers
Very minor and not sure if it is intentional, but in your ug you mentioned address accepts Words with or without spaces. However I can input a single digit integer, which is not very meaningful. 

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: neyapraveen/ped#6

### PR DESCRIPTION
**Pull request by wasjoe1**
Let's make improvements in our UG to tackle bugs found during PED.

- Specify definition of words
- Include clear feature, clear guest and clear vendor will be implemented in the future
- Mentions that a person can be recorded as both a guest & vendor
- Rectify error messages

resolves #214
resolves #209
resolves #192

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -551,37 +569,38 @@ Acceptable values for `e/EMAIL`:
         - end with a domain label at least 2 characters long.
         - have each domain label start and end with alphanumeric characters.
         - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.
-- Error message for invalid email: "Emails should contain two segments separated by an @ symbol."
+- Error message for invalid email: "Invalid command format!" followed by an instruction on the proper usage of the command you were executing.
```
Comment (samuelim01) [inline docs/UserGuide.md:572]: Not sure about this, seems like the current error message is slightly different?
```
Emails should be of the format local-part@domain and adhere to the following constraints:
1. The local-part should only contain alphanumeric characters and these special characters, excluding the parentheses, (+_.-). The local-part may not start or end with any special characters.
2. This is followed by a '@' and then a domain name. The domain name is made up of domain labels separated by periods.
The domain name must:
    - end with a domain label at least 2 characters long
    - have each domain label start and end with alphanumeric characters
    - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -551,37 +569,38 @@ Acceptable values for `e/EMAIL`:
...
+- Error message for invalid RSVP status: “RsvpStatus Status should only have one of three values. yes, no, or unknown.”.
 
 Acceptable values for `d/DIETARY_REQUIREMENT`:
 - Alphanumeric word with or without spaces.
 
 Acceptable values for `tn/TABLE_NUMBER`:
 - Non-negative integer with no spaces or special characters.
 - Preceding zeros will be trimmed.
-- Error message for invalid table numbers: "Table numbers should be positive numbers with no spaces or special characters."
+- Error message for invalid table numbers: "Table numbers should only contain numbers."
```
Comment (samuelim01) [inline docs/UserGuide.md:592]: Seems like the current error message has no full stop 😢 
```suggestion
- Error message for invalid table numbers: "Table numbers should only contain numbers"
```
image

File: docs/UserGuide.md [hunk 3]
```diff
@@ -551,37 +569,38 @@ Acceptable values for `e/EMAIL`:
...
 
 Acceptable values for `t/TAG`:
 - Alphanumeric word without spaces.
-- Error message for invalid tags: "Tags should have no spaces or special characters."
+- Error message for invalid tags: "Tag names should be alphanumeric"
```
Comment (samuelim01) [inline docs/UserGuide.md:596]:
```suggestion
- Error message for invalid tags: "Tags names should be alphanumeric"
```

image

File: docs/UserGuide.md [hunk 4]
```diff
@@ -551,37 +569,38 @@ Acceptable values for `e/EMAIL`:
...
 - `yes`
 - `no`
 - `unknown`
 - Inputs with no values when adding a guest (e.g. `r/`) signify that RSVP status should be stored as `unknown`.
 - Inputs with no values when filtering guests (e.g. `r/`) signify that you wish to filter for guests with `unknown` RSVP status.
-- Error message for invalid RSVP status: “RSVP status can only be `yes`, `no` or `unknown`”.
+- Error message for invalid RSVP status: “RsvpStatus Status should only have one of three values. yes, no, or unknown.”.
 
 Acceptable values for `d/DIETARY_REQUIREMENT`:
 - Alphanumeric word with or without spaces.
```
Comment (samuelim01) [inline docs/UserGuide.md:587]:
```suggestion
- Alphanumeric word with or without spaces and should not be blank.
- Error message for invalid dietary requirement: “Dietary requirements should only contain alphanumeric characters and spaces, and it should not be blank”.
```

image

File: docs/UserGuide.md [hunk 5]
```diff
@@ -533,13 +552,12 @@ Format: `exit`
...
 - Alphanumeric word with or without spaces and should not be blank.
-- Error message for invalid name: “Names cannot be blank and should not contain any special characters”.
+- Error message for invalid name: “Names should only contain alphanumeric characters and spaces, and it should not be blank”.
 
 Acceptable values for `p/PHONE`:
 - Numbers with no spaces or special characters.
 - At least 3 numbers.
-- Error message for invalid phone number: “Phone numbers should contain only numbers, with no spaces or special characters”.
-
+- Error message for invalid phone number: "Invalid command format!" followed by an instruction on the proper usage of the command you were executing.
```
Comment (samuelim01) [inline docs/UserGuide.md:560]: Not too sure why the error message for this was changed! 
image

File: docs/UserGuide.md [hunk 6]
```diff
@@ -551,37 +569,38 @@ Acceptable values for `e/EMAIL`:
...
 --------------------------------------------------------------------------------------------------------------------
 
 ### 6.2. Appendix B: Miscellaneous error messages
 
 User input is completely invalid (e.g. `abc` or `vsdf`):
-- Display error message "No such command exists".
+- Display error message "Unknown command.".
 
 User input begins with `vendor` or `guest`, but does not include a valid command word (e.g. `vendor abc` or `guest adddd`):
 - Display error message "Please specify a command".
```
Comment (samuelim01) [inline docs/UserGuide.md:606]: Could you modify these two lines too?
image

File: docs/UserGuide.md [hunk 7]
```diff
@@ -64,6 +64,8 @@ WedLog is a desktop app for wedding planning, optimized for use via a Command Li
 
 **Notes about the command format:** <br />
 
+* Words are defined to be any letters, numbers or special characters of length 1 or more
```
Comment (samuelim01) [inline docs/UserGuide.md:67]: Based on #214, I believe this still means that users who look the section on Address without seeing this small note will still perceive the single digit integer to be not very meaningful. Is it possible to just specify the acceptable values for Address directly as "any non blank values", or something along those lines? We can still keep this definition here for the purposes of the "alphanumeric words" of names, etc.

**Review by samuelim01 — CHANGES_REQUESTED** [review]
Thanks for the hard work :) Just a few nits!

**Comment by samuelim01** [comment]
Oh by the way, if you want your PR to automatically close the issues your mentioned, I think you left out the closing keywords

File: docs/UserGuide.md [hunk 8]
```diff
@@ -34,7 +34,7 @@ WedLog is a desktop app for wedding planning, optimized for use via a Command Li
     &emsp; 2.6.3 [Redoing last action: `redo`](#2-6-3-redoing-last-action-redo)<br>
     &emsp; 2.6.4 [Exiting the program: `exit`](#2-6-4-exiting-the-program-exit)<br>
 3. [FAQ](#3-faq)
-4. [Known Issues](#4-known-issues)
+4. [Known Issues/ Future implementations](#4-known-issues-future-implementations)
```
[the remark this answers is not in the document]
Comment (wasjoe1) [reply docs/UserGuide.md:37]: Ohh was thinking of this alternative too. Not sure if dedicating an entire section for 1 line each makes it less readable, though I will just edit to different sections first and see what the rest think as well!

File: docs/UserGuide.md [hunk 9]
```diff
@@ -120,8 +124,8 @@ Expected behaviour upon success:
 - Displays a message showing the added guest.
 
 Expected behaviour upon failure:
-- As `NAME` is a compulsory parameter for guests, not providing this parameter would result
-in the error message “Please specify the guest’s name using the format `n/NAME`”.
+- As `NAME` is a compulsory parameter for vendors, not providing this parameter would result
+  in the error message “Invalid command format” followed by instruction on guest add usage.
```
[the remark this answers is not in the document]
Comment (wasjoe1) [reply docs/UserGuide.md:128]: Resolved!

File: docs/UserGuide.md [hunk 10]
```diff
@@ -34,7 +34,7 @@ WedLog is a desktop app for wedding planning, optimized for use via a Command Li
     &emsp; 2.6.3 [Redoing last action: `redo`](#2-6-3-redoing-last-action-redo)<br>
     &emsp; 2.6.4 [Exiting the program: `exit`](#2-6-4-exiting-the-program-exit)<br>
```
[the remark this answers is not in the document]
Comment (wasjoe1) [reply docs/UserGuide.md:35]: Resolved!

File: docs/UserGuide.md (hunk 1 above)
Comment (wasjoe1) [reply docs/UserGuide.md:572]: Resolved!

File: docs/UserGuide.md (hunk 2 above)
Comment (wasjoe1) [reply docs/UserGuide.md:592]: Resolved!

File: docs/UserGuide.md (hunk 3 above)
Comment (wasjoe1) [reply docs/UserGuide.md:596]: Resolved!

File: docs/UserGuide.md (hunk 4 above)
Comment (wasjoe1) [reply docs/UserGuide.md:587]: Resolved!

File: docs/UserGuide.md (hunk 6 above)
Comment (wasjoe1) [reply docs/UserGuide.md:606]: Resolved!

File: docs/UserGuide.md (hunk 5 above)
Comment (wasjoe1) [reply docs/UserGuide.md:560]: Resolved!

**Review by samuelim01 — APPROVED** [review]
LGTM! 😄

**Pull request merged**